modules/3d/include/opencv2/3d.hpp

```diff
@@ -2523,15 +2523,15 @@ class CV_EXPORTS Octree {
     *
     * @param maxDepth The max depth of the Octree. The maxDepth > -1.
     */
-    explicit Octree(int maxDepth);
+    explicit Octree(size_t maxDepth);
 
     /** @overload
-    * @brief Create an Octree from the PointCloud data with the specific max depth.
+    * @brief Create an Octree from the PointCloud data with the specific resolution.
     *
     * @param pointCloud Point cloud data.
-    * @param maxDepth The max depth of the Octree.
+    * @param resolution The size of the octree leaf node.
     */
-    Octree(const std::vector<Point3f> &pointCloud, int maxDepth);
+    Octree(const std::vector<Point3f> &pointCloud, double resolution);
```

Comment:
maybe both resolution and maxDepth could be specified or one of them; why remove maxDepth?

Reply:
Perhaps it's reasonable to retain both options.

The resolution denotes the size of the leaf node. I opted for resolution as it aligns well with point cloud compression, where preserving information at this precision is sufficient.

For the current octree functions, such as KNN search, it might be better to retain both methods. Simultaneously, the two can be calculated from one to the other.

What's your opinion?

Reply:
Since octree is used not only for point cloud compression, I'd propose you to keep both constructors (resolution and maxDepth)